src/main/java/com/team766/robot/mechanisms/januaryTag.java

```diff
@@ -0,0 +1,208 @@
+     * @return Transform3d - the Transform3d from the best target that the camera is currently tracking
+     */
+
+    public Transform3d getBestCameraToTarget(PhotonTrackedTarget target){
+        return target.getBestCameraToTarget();
+    }
+
+    /*
+     * This method returns the data from the best target that the camera is currently tracking, not the Transform3d data
+     * @return ArrayList<Double> - the data from the best target that the camera is currently tracking
+     * [0] is the yaw
+     * [1] is the pitch
+     * [2] is the area that the apriltag fills the camera view with
+     * [3] is the fiducial id of the april tag
+     */
+    public ArrayList<Double>  getPhotonTrackedTargetData(){
+        ArrayList<Double> arr = new ArrayList<Double>();
+        PhotonTrackedTarget theTarget = getBestTrackedTarget();
+        arr.add(theTarget.getYaw());
+        arr.add(theTarget.getPitch());
+        arr.add(theTarget.getArea());
+        arr.add((double) theTarget.getFiducialId());
+        return arr;
+    }
+
+    /*
+     * This is a method to return data given from a Transform3d
+     * @return an ArrayList<Double> with values about the Transform3d, assuming the camera is (0,0)
+     * [0] is the x value of the target relative to the camera
+     * [1] is the y value of the target relative to the camera
+     * [2] is the z value of the target relative to the camera
+     */
+    public ArrayList<Double> getTransform3dData(){
+        ArrayList<Double> arr = new ArrayList<Double>();
+        Transform3d targetTransform = getBestCameraToTarget(getBestTrackedTarget());
+        arr.add(targetTransform.getX());
+        arr.add(targetTransform.getY());
+        arr.add(targetTransform.getZ());
+        return arr;
+    }
+    
+    public void setDeadzoneX(double dz){
+        deadzoneX = dz;
+    }
+
+    public void setDeadzoneY(double dz){
+        deadzoneY = dz;
+    }
+
+    public void setDeadzones(double x, double y){
+        deadzoneX = x;
+        deadzoneY = y;
+    }
+    /*
+     * This method is the method that you should call to make the robot go to the target. Currently, it will go to any apriltag that it sees, but we can change that later.
+     */
+    public void go(){
+        
+        while(2>1){
+            
+            Transform3d targetTransform = getBestCameraToTarget(getBestTrackedTarget());
+            Transform3d scoring = targetTransform.plus(offset);
+
+            if(scoring.getX() + deadzoneX > 0 && scoring.getX() - deadzoneX < 0 && scoring.getY() + deadzoneY > 0 && scoring.getY() - deadzoneY < 0){
+                log("Outtaking");
+                break;
+            }else{
+                
+
+                double x_scoring = scoring.getX();
+                double y_scoring = scoring.getY(); 
+
+                double x_target = x_scoring + Math.cos(scoring.getX()) * (x_targetConstant * y_scoring);
```

Comment:
```suggestion
                double yaw_scoring = scoring.getRotation().getZ();
                double x_target = x_scoring + Math.cos(yaw_scoring) * (x_targetConstant * y_scoring);
```

It looks like, instead of calling the three rotation angles `roll`, `pitch`, and `yaw`, they instead name these angles by their rotation axis. Yaw is rotation around the Z (vertical) axis
https://github.com/wpilibsuite/allwpilib/blob/5fac18ff4ad1db92ec6fcbd437043a38028e99c4/wpimath/src/main/java/edu/wpi/first/math/geometry/Rotation3d.java#L335-L340